### PR TITLE
ODS API querying helpers

### PR DESCRIPTION
**Pull request by mrob95**
* **Please check if the Pull Request fulfills these requirements**
- [x] The commit message is clear and concise
- [x] Test for the changes have been added and reviewed
- [x] Documentation has been added to all new features and edited code has had documentation reviewed

* **What kind of change does this Pull request introduce?** (Bug fix, feature, docs update, ...)

Adds a couple of new functions for querying the [ODS API](https://digital.nhs.uk/services/organisation-data-service/guidance-for-developers/organisation-endpoint). The main one, `ODS_lookup.get_addresses`, can be used to retrieve name and address information for a list of ODS codes as a pandas data frame. e.g.
```
>>> get_addresses(pd.Series(["X26"]))
              Org_AddrLn1 Org_Code Org_Country     Org_Name Org_PostCode Org_Town
    0  1 TREVELYAN SQUARE      X26     ENGLAND  NHS Digital      LS1 6AE    LEEDS
```
	 
The api only lets you query for a single code at a time. I don't *think* making serial requests is going to cause a major performance issue but it might for very large requests. Could investigate async requests if necessary.

### CODE REVIEW

File: codonPython/ODS_lookup.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        org_name = org_info["Name"]
+        org_address = org_info["GeoLoc"]["Location"]
+        result = {
+            "Org_Code": code,
+            "Org_Name": org_name.title().replace("Nhs", "NHS"),
+            **{f"Org_{k}": v for k, v in org_address.items() if k != "UPRN"},
+        }
+        return result
+
+    to_query = set(codes)
```
[the remark this answers is not in the document]
Comment (mrob95) [reply codonPython/ODS_lookup.py]: Do you think this needs a comment?

File: codonPython/ODS_lookup.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    Parameters
+    ----------
+    code : str
+        3 character organization code.
+
+    Returns
+    ----------
+    dict
+        The data returned from the API.
```
Comment (murd0) [inline codonPython/ODS_lookup.py]: Can I have an example here? You can comment it out

File: codonPython/ODS_lookup.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import requests
+from typing import Dict, Iterable, Callable, List, Optional
+import pandas as pd
+import numpy as np
+
+
+def query_api(code: str) -> Dict:
```
[the remark this answers is not in the document]
Comment (mrob95) [reply codonPython/ODS_lookup.py]: Do you think that `get_addresses` should be renamed as well? `addresses_from_ods_codes` or something would be more descriptive but slightly wordy. This is all namespaced under `ODS_lookup` so maybe not necessary?

[the remark this answers is not in the document]
Comment (murd0) [reply codonPython/ODS_lookup.py]: On second thoughts, change it back. intended usage being `from codonPython import ODS_Lookup` the namespace is clear

**Pull request merged**